apps/ledger-live-desktop/src/renderer/components/SyncOnboarding/Manual/EarlySecurityChecks/index.tsx

```diff
@@ -0,0 +1,290 @@
+import { Status as SoftwareCheckStatus } from "../types";
+import { getDeviceModel } from "@ledgerhq/devices";
+import { useSelector } from "react-redux";
+import { urls } from "~/config/urls";
+import { openURL } from "~/renderer/linking";
+import { localeSelector } from "~/renderer/reducers/settings";
+import { setDrawer } from "~/renderer/drawers/Provider";
+import UpdateFirmwareModal, {
+  Props as UpdateFirmwareModalProps,
+} from "~/renderer/modals/UpdateFirmwareModal";
+import { Device } from "@ledgerhq/live-common/hw/actions/types";
+import { initialStepId } from "~/renderer/screens/manager/FirmwareUpdate";
+import GenuineCheckErrorDrawer, {
+  Props as GenuineCheckErrorDrawerProps,
+} from "./GenuineCheckErrorDrawer";
+import DeviceNotGenuineDrawer, {
+  Props as DeviceNotGenuineDrawerProps,
+} from "./DeviceNotGenuineDrawer";
+import { useTranslation } from "react-i18next";
+
+export type Props = {
+  onComplete: () => void;
+  device: Device;
+  /** display the genuine check as validated but still check it */
+  optimisticGenuineCheck: boolean;
+  restartChecksAfterUpdate: () => void;
+};
+
+const commonDrawerProps = {
+  forceDisableFocusTrap: true,
+  preventBackdropClick: true,
+};
+
+/**
+ * Component representing the early security checks step, which polls the current device state
+ * to display correctly information about the onboarding to the user
+ */
+const EarlySecurityChecks = ({
+  onComplete,
+  device,
+  optimisticGenuineCheck,
+  restartChecksAfterUpdate,
+}: Props) => {
+  const { t } = useTranslation();
+  const locale = useSelector(localeSelector);
+  const whySecurityChecksUrl =
+    locale in urls.genuineCheck
+      ? urls.genuineCheck[locale as keyof typeof urls.genuineCheck]
+      : urls.genuineCheck.en;
+
+  const [genuineCheckStatus, setGenuineCheckStatus] = useState<SoftwareCheckStatus>(
+    optimisticGenuineCheck ? SoftwareCheckStatus.optimisticCompleted : SoftwareCheckStatus.inactive,
+  );
+  const [firmwareUpdateStatus, setFirmwareUpdateStatus] = useState<SoftwareCheckStatus>(
+    SoftwareCheckStatus.inactive,
+  );
+  const [availableFirmwareVersion, setAvailableFirmwareVersion] = useState<string>("");
+  const [updateInterrupted, setUpdateInterrupted] = useState(false);
+
+  const deviceId = device.deviceId ?? "";
+  const deviceModelId = device.modelId;
+  const productName = getDeviceModel(device.modelId).productName;
+
+  const genuineCheckActive =
+    genuineCheckStatus === SoftwareCheckStatus.active ||
+    genuineCheckStatus === SoftwareCheckStatus.optimisticCompleted;
+
+  const {
+    genuineState,
+    error: genuineCheckError,
+    devicePermissionState,
+    resetGenuineCheckState,
+  } = useGenuineCheck({
+    getGenuineCheckFromDeviceId,
+    isHookEnabled: genuineCheckActive,
+    deviceId,
+  });
+
+  const {
+    state: { deviceInfo, firmwareUpdateContext: latestFirmware, status, lockedDevice },
+  } = useGetLatestAvailableFirmware({
```

Comment:
You can also retrieve an `error` object from this hook: it can be good to set `firmwareUpdateStatus` to `failed` if an error occurred, and the user has a button to continue

There is one thing to pay attention to: some error are retried by the inner device action, and so the action has not yet failed completely
You can check for this with the following:

```TS
      // If the current error triggered a retry attempt, does not display failure
      if (
        getLatestAvailableFirmwareError &&
        !(
          getLatestAvailableFirmwareError.type === "SharedError" &&
          getLatestAvailableFirmwareError.retrying
        )
      ) {
        log(
          "EarlySecurityCheck",
          "Failed to retrieve latest firmware version with error:",
          getLatestAvailableFirmwareError.name,
        );

        // Or equivalent
        setFirmwareUpdateCheckStatus("failed");
      }
```

If the error has a `type` of `SharedError`, then you can check if it has triggered a retry with the `retrying` property ;)